### DISCUSSION TITLE
When building an index, the program will be stuck all the time and no error will be reported. Is it a problem to wait or to set parameters?

### DISCUSSION THREAD

**Discussion by sky186** [milvus-io/milvus#4868] [Q&A and General discussion]
感觉是数据量太大，这个是1200w 2048 维度的特征，创建索引会出现这个问题
同样数据量的 512维特征，运行很正常，搜索正确

index_file 设置不能太大 设置的32
nlist 4096

**Comment by yhmo** [comment]
When create_index() is called, it will blocked until all segments finish build index.
I just did some test: 40W 2048 dim vectors, index_file_size=32, it cost 1 minute to build index on my machine(12 cores cpu). So, 1200w 2048 dim vectors could cost half hour to build index.

**Comment by shengjun1985** [comment]
> index_file 设置不能太大 设置的32

Why?
Then, then row count of each segment is about (32 * 1024 * 1024) / (2048 * 4) = 4096.
Too much small segments will cause poor performance.
